maintenance/wikia/HttpsMigration/migrateCustomCss.php

```diff
@@ -0,0 +1,328 @@
+
+/**
+ * Class MigrateCustomCssToHttps
+ */
+class MigrateCustomCssToHttps extends Maintenance {
+
+	protected $saveChanges  = false;
+	protected $fh;	// handle to the output csv file
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Migrates urls in custom CSS assets to HTTPS';
+		$this->addOption( 'saveChanges', 'Edit articles for real.', false, false, 'd' );
+		$this->addOption( 'file', 'CSV file where to save values that are going to be altered', false, true, 'f' );
+	}
+
+	public function __destruct()  {
+		if ( $this->fh ) {
+			fclose( $this->fh );
+		}
+	}
+
+	/**
+	 * Logs urls value change if the new value is different than the old one.
+	 */
+	public function logUrlChange( $description, $oldValue, $newValue ) {
+		global $wgCityId;
+		if ( $oldValue !== $newValue ) {
+			if ( $this->fh ) {
+				fputcsv( $this->fh,
+					[ $wgCityId, $this->currentTitle->getDBkey(), $description, $oldValue, $newValue ] );
+			}
+		}
+	}
+
+	/**
+	 * Returns css file edit summary
+	 */
+	private function getEditSummary() {
+		return "Applying changes that should make this CSS file HTTPS-ready. If you have any questions or noticed " .
+			"issues related to this edit, please reach out to us using the [[Special:Contact]] page.";
+	}
+
+	private function isHttpUrl( $url ) {
+		return startsWith( $url, 'http://' );
+	}
+
+	private function isHttpsUrl( $url ) {
+		return startsWith( $url, 'https://' );
+	}
+
+	private function isProtocolRelativeUrl( $url ) {
+		return startsWith( $url, '//' );
+	}
+
+	/**
+	 * True if the url doesn't contain the host part.
+	 */
+	private function isLocalUrl( $url ) {
+
+		if ( $this->isHttpUrl( $url ) ||
+			$this->isHttpsUrl( $url ) ||
+			$this->isProtocolRelativeUrl( $url ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	private function isWikiaComSubdomainUrl( $url ) {
+		$host = parse_url( $url, PHP_URL_HOST );
+		return endsWith( $host, '.wikia.com' );
+	}
+
+	/**
+	 * True the the url points outside of our wikia.com and vignette domain.
+	 */
+	private function isThirdPartyUrl( $url ) {
+		$host = parse_url( $url, PHP_URL_HOST );
+		return !endsWith( $host, '.wikia.com' ) && !endsWith( $host, '.wikia.nocookie.net' );
+	}
+
+	/**
+	 * Returns true if the url points to one of many variations of our image domains.
+	 * Handles a lot of historical formats that are not used anymore.
+	 */
+	private function isWikiaImageUrl( $url ) {
+		$host = parse_url( $url, PHP_URL_HOST );
+		return ( preg_match( '/^(www\.)?(vignette|images|image|img|static|slot)\d*\.wikia\.nocookie\.net$/', $host ) ||
+			preg_match( '/^(www\.)?slot\d*[\.-]images\.wikia\.nocookie\.net$/', $host ) ||
+			$host === 'images.wikia.com' ||
+			$host === 'static.wikia.com' );
+	}
+
+	/**
+	 * Replaces http protocl with https. Protocol-relative urls are not affected.
+	 */
+	private function upgradeToHttps( $url ) {
+		if ( $this->isHttpUrl( $url ) ) {
+			$url = http_build_url( $url, [ 'scheme' => 'https' ] );
```

Comment:
@mech I know this is very unlikely but in case this function return `false` we shouldn't probably change anything and log this error.
There are more cases like this

Reply:
I think you meant to ping @themech :P

Reply:
LGTM  👍 .. please fix and deploy it.. tks

Reply:
@mech sorry!

Reply:
added basic error handling in https://github.com/Wikia/app/pull/14908/commits/f9a610ecbcb2166adcc6d4d083af8cf6fae3d81e